Documentation/tutorial/validate your markdown files.md

```diff
@@ -0,0 +1,128 @@
+    * `{0}` the name of tag.
+    * `{1}` the whole tag.
+    For example, the `messageFormatter` is `{0} is the tag name of {1}.`, and the tag is `<H1 class="heading">` match the rule, then it will output following message: `H1 is the tag name of <H1 class="heading">.`
+4.  `customValidatorContractName` is an extension tag rule contract name for complex validation rule, see [How to create a custom html tag validator](#how-to-create-a-custom-html-tag-validator).
+5.  `openingTagOnly` is a boolean, if `true`, it will only apply to opening tag, e.g. `<H1>`, otherwise, it will also apply to closing tag, e.g. `</H1>`.
+
+### How to create a custom html tag validator
+1.  Create a project in your code editor (e.g. visual studio).
+2.  Add nuget package `Microsoft.DocAsCode.Plugins` and `Microsoft.Composition`.
+3.  Create a class and implements @Microsoft.DocAsCode.Plugins.ICustomMarkdownTagValidator.
+4.  Add ExportAttribute with contract name.
+
+For example, we require tags should contain text `class`:
+```csharp
+[Export("Require text class", typeof(ICustomMarkdownTagValidator))]
+public class MyMarkdownTagValidator : ICustomMarkdownTagValidator
+{
+    public bool Validate(string tag)
+    {
+        return tag.Contains("class");
+    }
+}
+```
+
+## Customize token validation rules
+In `rules` property, we can add any [validation plug-in](#how-to-create-a-plug-in-for-markdown-validation) by following style:
+```json
+{
+   "rules": [ { "name": "rule name", "disable": false } ]
+}
+```
+or simplify it to:
+```json
+{
+    "rules": [ "rule name" ]
+}
+```
+
+### With contract name `default`
+Any rules with contract name `default` will be enabled by default. This means when no `md.style` file, it is same as a `md.style` file with following content:
+```json
+{
+   "rules": [ { "name": "default", "disable": false } ]
+}
+```
+And you can disable default rules by following config:
+```json
+{
+   "rules": [ { "name": "default", "disable": true } ]
+}
+```
+
+### How to create a plug-in for markdown validation
+1.  Create a project in your code editor (e.g. visual studio).
+2.  Add nuget package `Microsoft.DocAsCode.MarkdownLite` and `Microsoft.Composition`.
+3.  Create a class and implements @Microsoft.DocAsCode.MarkdownLite.IMarkdownTokenValidatorProvider
+
+    > [!TIP]
+    > @Microsoft.DocAsCode.MarkdownLite.MarkdownTokenValidatorFactory contains some help methods to create a validator.
+4.  Add ExportAttribute with contract name, and if contract name is `default` it will apply automatically.
+
+
+For example, we require all code block with language `csharp`:
+```csharp
+[Export("default", typeof(IMarkdownTokenValidatorProvider))]
+public class MyMarkdownTokenValidatorProvider : IMarkdownTokenValidatorProvider
+{
+    public ImmutableArray<IMarkdownTokenValidator> GetValidators()
+    {
+        return ImmutableArray.Create(
+            MarkdownTokenValidatorFactory.FromLambda<MarkdownCodeBlockToken>(t =>
```

Comment:
I see. Can we add more explanation here so user could understand
1. where they can find the tokens already supported. I found those by search for 'token' in docfx reference document.
2. how they can implement their own token. This is more complex, it might need a separate documentation for helping user contribute to docfx engine.